lib/telepoison.ex

```diff
@@ -35,20 +35,67 @@ defmodule Telepoison do
     :otel_propagator_text_map.inject(headers)
   end
 
-  def request(%Request{options: opts} = request) do
-    save_parent_ctx()
-    span_name = Keyword.get_lazy(opts, :ot_span_name, fn -> compute_default_span_name(request) end)
+  @doc """
+  Performs a request using Telepoison with the provided `request` options.
+
+  See `HTTPoison.request/1` for further details regarding `request` options.
+
+  Will attempt to automatically set the `http.route` Open Telemetry metadata attribute by
+  removing the last part of the `request.url`, since this part typically contains dynamic data.
+
+  If this behavior is not desirable, it can be set directly by using the `:resource_route` optional parameter.
+
+    ## Examples
+
+      iex> request = %HTTPoison.Request{
+      ...> method: :post,
+      ...> url: "https://www.example.com/users/edit/2",
+      ...> body: "{\"foo\": 3}",
+      ...> headers: [{"Accept", "application/json"}]}
+      iex> Telepoison.request(request, resource_route: "users/edit/:user_id")
+      {:ok, response}
+
+  """
+  def request(%Request{options: request_opts} = request, opts \\ []) do
+    span_name = Keyword.get_lazy(request_opts, :ot_span_name, fn -> compute_default_span_name(request) end)
+
+    [env: env, service: service] = Logger.metadata() |> Keyword.take([:service, :env])
+
+    path =
+      case Keyword.get(opts, :resource_route) do
+        nil ->
+          request.url
+          |> URI.parse()
+          |> then(fn uri ->
+            uri.path
+            |> String.split("/")
+            # Skips the first item which is an empty string and the last entry which may contain dynamic data
+            # e.g. user id etc.
+            |> Enum.slice(1..-2)
+            |> Enum.join("/")
+            |> (&"#{&1}/:subpath").()
+          end)
```

Comment:
Split into its own function, e.g. `infer_resource_from_path(`, and add some tests for it?

```elixir
assert infer_resource_from_path("/user") == "/user"
assert infer_resource_from_path("/user/12") == "/user/:subpath"
assert infer_resource_from_path("/user/12/details") == "/user/:subpath"
assert infer_resource_from_path("/") == "/"
```

Also, I don't think only the last entry may contain dynamic data. Ultimately it's a tradeoff how "good" the inferred route would be vs the risk of having dynamic params in there. I suggested that just taking the first path component is very unlikely to contain dynamic parameters (e.g. noone uses non-prefixed routes that start directly with a parameter like `/12/details`) but there may be better heuristics.

Reply:
- Updated README.md
- Added tests
- Updated the `:resource_route` inference to essentially grab the first slug on the URL (if present! The above scenarios pass are covered by the tests)
- Moved the `:resource_route` inference logic to its own module